rust/agama-network/src/nm/client.rs

```diff
@@ -328,17 +331,47 @@ impl<'a> NetworkManagerClient<'a> {
         // FIXME: Do not like that an activation/deactivation could not apply the changes because of
         // a roolback when calling this method with an error.
         if conn.is_up() {
+            if let Some(interface) = &conn.interface {
+                for device in devices {
+                    if interface.to_string() == device.name {
+                        if let Some(device_conn) = device.connection {
```

Comment:
JFYI, in the future it would be possible to combine `if let` and `&&` expressions. See https://blog.rust-lang.org/2025/06/26/Rust-1.88.0/#let-chains.

Reply:
We need rust version 2024 isn't it?